int odrive_index
string state
